jobbergate-api/jobbergate_api/apps/applications/routers.py

```diff
@@ -95,12 +99,19 @@ async def applications_upload(
             status_code=status.HTTP_413_REQUEST_ENTITY_TOO_LARGE,
             detail=message,
         )
-    s3man_applications[application_id] = upload_file.file
+
+    write_application_files_to_s3(application_id, upload_files, remove_previous_files=True)
```

Comment:
There is no need because `write_applications_files_to_s3` only writes what it is prepared to (application source file and templates): https://github.com/omnivector-solutions/jobbergate/blob/667186b3538f3d1c28e2f77563d2af51f8ed8756/jobbergate-api/jobbergate_api/s3_manager.py#L65